e2e/test/helpers/TestDevice.cs

```diff
@@ -27,14 +27,14 @@ public class TestDevice : IDisposable
     {
         private static readonly SemaphoreSlim s_semaphore = new(1, 1);
 
-        private static readonly IRetryPolicy s_createRetryPolicy = new HubServiceTestRetryPolicy();
+        private static readonly IIotHubServiceRetryPolicy s_createRetryPolicy = new HubServiceTestRetryPolicy();
 
         private static readonly HashSet<IotHubServiceErrorCode> s_getRetryableStatusCodes = new()
         {
             IotHubServiceErrorCode.DeviceNotFound,
             IotHubServiceErrorCode.ModuleNotFound,
         };
-        private static readonly IRetryPolicy s_getRetryPolicy = new HubServiceTestRetryPolicy(s_getRetryableStatusCodes);
+        private static readonly IIotHubServiceRetryPolicy s_getRetryPolicy = new HubServiceTestRetryPolicy(s_getRetryableStatusCodes);
```

Comment:
We can get rid of the custom retry policy here and just use the new built-in one.

Reply:
Oh, I'm wrong about that, because we add the extra codes, but we can change the HubServiceTestRetryPolicy to inherit from the new ones you've brought for Hub service.

Reply:
The problem is that the built-in ones cannot take StatusCode and use it to determine if it should retry or not, so we still need a custom retry policy here, at least for the hub service.